src/main/java/com/bigchaindb/builders/BigchainDbTransactionBuilder.java

```diff
@@ -271,6 +273,18 @@ public ITransactionAttributes addInput(String fullfillment, FulFill fullFill, Ed
             return this;
         }
 
+        @Override
+        public ITransactionAttributes addInput(Details fullfillment, FulFill fullFill, EdDSAPublicKey... publicKeys) {
+            for (EdDSAPublicKey publicKey : publicKeys) {
+                Input input = new Input();
+                input.setFullFillment(fullfillment);
+                input.setFulFills(fullFill);
+                input.addOwner(KeyPairUtils.encodePublicKeyInBase58(publicKey));
+                this.inputs.add(input);
+            }
+            return this;
```

Comment:
code that adds the fulfillment to the input, which now knows how to accept `Details`

Reply:
accepted